### PR TITLE
feat(FEC-13691): Application events analytics reporting to kava

### PR DESCRIPTION
**Pull request by JonathanTGold**
### Description of the Changes

Add Application events analytics reporting to kava

Solves FEC-13691

#### Related PRs:
https://github.com/kaltura/playkit-js-share/pull/40
https://github.com/kaltura/playkit-js-downloads/pull/43
https://github.com/kaltura/playkit-js-dual-screen/pull/131
https://github.com/kaltura/playkit-js-playlist/pull/56
https://github.com/kaltura/playkit-js-navigation/pull/352
https://github.com/kaltura/playkit-js-transcript/pull/180
https://github.com/kaltura/playkit-js-bumper/pull/111
https://github.com/kaltura/playkit-js-skip/pull/22

### CODE REVIEW

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]
note the type errors

File: src/application-events-model.ts [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+      buttonName: 'Dual__screen_change_layout',
+      buttonType: ButtonType.Unknown,
+      buttonValue: payload.layout
+    })
+  },
+  [ShareEvents.SHARE_CLICKED]: {
+    type: 'SHARE_CLICKED',
+    getEventModel: (): any => ({
+      eventType: ApplicationEventType.BUTTON_CLICKED,
+      buttonName: 'Share_embed_open',
```
Comment (SivanA-Kaltura) [inline src/application-events-model.ts:46]: button names should be consts

File: src/application-events-model.ts [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+    })
+  },
+  [NavigationEvents.NAVIGATION_SEARCH]: {
+    type: 'NAVIGATION_SEARCH',
+    getEventModel: (payload: any): any => {
+      const model = {
+        eventType: ApplicationEventType.BUTTON_CLICKED,
+        buttonType: ButtonType.Search
+      };
+      //navigation_search {"searchQuery":"ddd","activeTab":"All","availableTabs":["Slide"],"totalResults":0}  #@#@#
```
Comment (SivanA-Kaltura) [inline src/application-events-model.ts:181]: is the commented part still relevant ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/application-events-model.ts (hunk 1 above)
Comment (JonathanTGold) [reply src/application-events-model.ts:46]: why

Comment (MosheMaorKaltura) [reply src/application-events-model.ts:46]: Since there is no reuse for those names, and they exist only in one place, here, lets leave it as is

**Review by MosheMaorKaltura — APPROVED** [review]

**Pull request merged**